client/src/storageManager.js

```diff
@@ -0,0 +1,167 @@
+import { LuigiClientBase } from './baseClass';
+import { helpers } from './helpers';
+
+const pendingOperation = new Map();
+
+/**
+ * StorageManager allows you to use browser local storage (key/value); every storage operation is sent to managed by Luigi core.
+ * The idea is that different micro frontends can share/persist items using local storage.
+ * Since all storage operations are asynchronous (send vent to Luigi core that will reply once operation is finished), all the methods return Promises.
+ * @name storageManager
+ */
+class StorageManager extends LuigiClientBase {
+  /** @private */
+  constructor() {
+    super();
+    this.storageEventProcessor = new StorageEventProcessor();
+    helpers.addEventListener('storage',
+      (evt, listenerId) => this.storageEventProcessor.processEvent(evt, listenerId)
+    );
+  }
+
+  /**
+   * Store an item for a specific key.
+   * @memberof storageManager
+   * @param {string} key key used to identify the value
+   * @param {Object} value item to store; object must be stringifyable
+   * @returns {Promise<void>} resolves an empty value when storage operation is over; it will launch an error if storage is no supported, value cannot be stringify or you are using a luigi reserved key
+   * @example
+   * LuigiClient.storageManager().setItem('keyExample','valueExample').then(() => console.log('Value stored'))
+   */
+  setItem(key, value) {
+    return new Promise((resolve, reject) => {
+      this.storageEventProcessor.execute(resolve, reject, 'setItem', {key, value});
+    });
+  }
+
+  /**
+   * Retrieve an item for a specific key.
+   * @memberof storageManager
+   * @param {string} key used to identify the value
+   * @returns {Promise<Object>} resolves item retrieved from storage; it will launch an error if storage is no supported
+   * @example
+   * LuigiClient.storageManager().getItem('keyExample').then((value) => console.log);
+   */
+  getItem(key){
+    return new Promise((resolve, reject) => {
+      this.storageEventProcessor.execute(resolve, reject, 'getItem', {key});
+    });
+  }
+
+  /**
+   * Remove an item for a specific key.
+   * @memberof storageManager
+   * @param {string} key used to identify the value
+   * @returns {Promise<Object>} resolves item just removed from storage; it will launch an error if storage is no supported or you are using a luigi reserved key
```

Comment:
```suggestion
   * @returns {Promise<Object>} resolves an item just removed from storage. It will launch an error if storage is not supported or if you are using a Luigi reserved key.
```